Dockerfile

```diff
@@ -0,0 +1,123 @@
+# FROM mcr.microsoft.com/devcontainers/javascript-node:1-18
+FROM mcr.microsoft.com/devcontainers/javascript-node:22
+
+# Install main dependencies and Docker in one step to reduce layers
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
```

Comment:
Do we actually need all of these packages? Codespaces should have a lot of these by default already - i.e. `git`, `cmake`, ...etc.

From a new Codespace:

```
@tyliec ➜ /workspaces/uipa (main) $ dpkg -l cmake
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version                 Architecture Description
+++-==============-=======================-============-=======================================
ii  cmake          3.16.3-1ubuntu1.20.04.1 amd64        cross-platform, open-source make system
```